### DISCUSSION TITLE
rejseplanen: API 1.0 to 2.0

### DISCUSSION THREAD

**Discussion by Pade2204** [KristjanESPERANTO/MMM-PublicTransportHafas#174] [General]
The Hacon/Hafas API seems to change version for rejseplanen.dk at 4th december 2024. Version 1.0 to 2.0. Will this MMM be updated to that or is it already compatible?

[https://labs.rejseplanen.dk/hc/da/articles/21554723926557-Om-API-2-0-nyt](https://labs.rejseplanen.dk/hc/da/articles/21554723926557-Om-API-2-0-nyt)

**Comment by KristjanESPERANTO** [comment]
This module uses [hafas-client](https://github.com/public-transport/hafas-client), which uses a hafas API. The API mentioned on the linked page is, in my opinion (after a quick glance), a different one. It could of course be that they will also switch off the hafas API after introducing the new API version. I would wait and see. If something changes in the hafas API, [hafas-client](https://github.com/public-transport/hafas-client) would have to be adapted.

I think that [MMM-Rejseplanen](https://github.com/poketcalulator/MMM-Rejseplanen) will need to be adapted after the API update.

**Comment by Pade2204** [comment]
Do you know a status of the danish rejseplanen api upgrade?
Is this module adapted? or maybe plans to be so?

**Reply by KristjanESPERANTO** [reply]
> Do you know a status of the danish rejseplanen api upgrade?

No, but as already suspected, the HAFAS API does not seem to be affected for the time being. It is not even mentioned in the page you linked. 

> Is this module adapted? 

No. MMM-PublicTransportHafas does not use the API you are talking about - it uses the HAFAS API and it still works with the profile for rejseplanen. I just tested it.

> or maybe plans to be so?

No. I see no reason why.


[MMM-Rejseplanen](https://github.com/poketcalulator/MMM-Rejseplanen) uses the API you are talking about, so if you absolutely want to use that API, please contact its maintainer.


If you have problems getting MMM-PublicTransportHafas to run, please open an issue :slightly_smiling_face:

**Comment by KristjanESPERANTO** [comment]
There is also a fork of MMM-Rejseplanen: https://github.com/MarkNygaard/MMM-Rejseplanen. It's maintainer, @MarkNygaard, seems very active at GitHub, maybe he can help.

**Reply by Pade2204** [reply]
Well, how do I get Danish Station IDs in MMM-PublicTransportHafas?
Im not seeing other than Kopenhagen Hbh... so not sure how to search in the script to find IDs.

I need StationID for Sluseholmen Metro and busses around there, if possible in this MMM.

**Reply by KristjanESPERANTO** [reply]
To query ID for the rejseplanen hafas API you have to run `npm run query rejseplanen`. And in the config you have to use the profile `rejseplanen`.

Try that chonfig:

```js
	{
		module: "MMM-PublicTransportHafas",
		position: "top_right",
		config: {
			hafasProfile: "rejseplanen",
			stationID: "8600626",
			stationName: "Sluseholmen St. (Metro)",
		}
	},
```

Do you have a suggestion on how to improve the instructions in the README so that you would be able to get it to work yourself?

**Reply by Pade2204** [reply]
I see what went wrong. I installed the https://github.com/raywo/MMM-PublicTransportHafas version, and was trying to get that to work.

I have now isntalled the correct one (your version), but ran into error with node and nvm not being updated. I tried to update but ruined my MM2 :(

I have an old RP3, but is planning to make a new installation on a RP5 next week, so maybe things are more updated to this module.

**Reply by KristjanESPERANTO** [reply]
Okay, good luck. If you have a problem, don't hesitate to open a new issue 🙂 

I'm closing this discussion because the API mentioned has nothing to do with this module.
